homeassistant/components/homekit_controller/binary_sensor.py

```diff
@@ -20,17 +20,23 @@
     conn = hass.data[KNOWN_DEVICES][hkid]
 
     def async_add_service(aid, service):
```

Comment:
In the competing PR, @MartinHjelmare suggested reducing the duplication on this method. Yours already has less duplication but wondering if something like this might be the way forward:

```python
ENTITY_TYPES = {
    'motion': HomeKitMotionSensor,
    'contact': HomeKitContactSensor,
}

def async_add_service(aid, service):
    entity_class = ENTITY_TYPES.get(service['stype'])
    if not entity_class:
        return False
    async_add_entities([entity_class(conn, info)], True)
    return True
```